Mosaics

vnewey opened the pull request:
Adding mosaicing code to test in Argo

vnewey committed:
Added make_mosaic_cli to the entry points

vnewey committed:
Added a function to mosaic cogs

vnewey committed:
Automatically update integration test validation results

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+
+    fs = s3fs.S3FileSystem(anon=True)
+    # Configure S3
+    configure_s3_access(cloud_defaults=True, aws_unsigned="--aws_unsigned")
```
`--aws_unsigned` is probably not a valid input; I think it needs to be True or False

vnewey replied:
modified the use of aws_unsigned to match intertidal

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+            with open(file_list_name, "w") as f:
+                for cog in cogs:
+                    cog = cog.replace(
+                        "dea-public-data-dev/",
```
Likely don't want to hard-code `dea-public-data-dev/`

vnewey replied:
We can look at doing this later.
I did also put in another hard coded replace, to enable it to work on s3 prod

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+                for cog in cogs:
+                    cog = cog.replace(
+                        "dea-public-data-dev/",
+                        "/vsicurl/https://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com/",
```
Probably don't want to hard-code `"/vsicurl/https://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com/"` either

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+import os
+
+
+def find_files(directory, pattern):
```
This doesn't seem to be used in the code

vnewey replied:
It is not. I will remove it. :)

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+    product_dir,
+    output_dir,
+    dataset_maturity="final",
+    sunglint="",
```
Get rid of `sunglint="",` probably, I don't think we need it

vnewey replied:
all references to sunglint have been removed. :)

robbibt commented on intertidal/mosaics.py:
```diff
@@ -0,0 +1,193 @@
+            recursive=True,
+        )
+
+    if len(sunglint) > 0:
```
Get rid of this I think

vnewey committed:
fixes to mosaic code

removed any reference to sunglint
modified use of aws_unsigned
Changed to work on prod s3 as well as dev

vnewey committed:
removed unused function

vnewey committed:
Automatically update integration test validation results
